### PR TITLE
Feature/445 implement accept reject backend

### PR DESCRIPTION
**Pull request by paola-quintanilla-jalafund**
[AB#](https://dev.azure.com/funjaladev/DevLevel3/_workitems/edit/445)

# Overview

Add Endpoint, service, and repository functions to update a Postulation

image

### CODE REVIEW

**Review by rodrigo-baldiviezo-jalafund — DISMISSED** [review]

**Review by niver-copa-jalafund — DISMISSED** [review]

**Review by lina-ramirez-jalafund — DISMISSED** [review]

File: tests/api/Jalasoft.TeamUp.Resumes.API.Tests/PatchPostulationApiTests.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            var request = this.mockHttpContext.Request;
+            request.Body = SetStream.Setstream(body);
+            Postulation postulation = null;
+            this.mockPostulationsService.Setup(service => service.PatchPostulation(It.IsAny<Postulation>())).Returns(postulation);
+            var response = await this.patchPostulation.Patch(request, 8);
+            var updatedResult = Assert.IsType<ObjectResult>(response);
+            Assert.Equal(404, updatedResult.StatusCode);
+        }
+
+        [Theory]
```
Comment (cristian-chavez-jalafund) [inline tests/api/Jalasoft.TeamUp.Resumes.API.Tests/PatchPostulationApiTests.cs]: please add more use cases or make these Fact instead of Theory

Comment (cristian-chavez-jalafund) [reply tests/api/Jalasoft.TeamUp.Resumes.API.Tests/PatchPostulationApiTests.cs]: the same for the others

File: src/dal/Jalasoft.TeamUp.Resumes.DAL/PostulationsRepository.cs [hunk 2]
```diff
@@ -66,15 +66,18 @@ public IEnumerable<Postulation> GetAllByProjectId(string projectId)
...
+            {
+                var sp = "Get_Postulation_By_Id";
+                var values = new { id = id };
+                postulation = db.QuerySingleOrDefault<Postulation>(sp, values, commandType: CommandType.StoredProcedure);
+            }
 
-        public IEnumerable<Resume> GetByName(string name)
-        {
-            throw new NotImplementedException();
+            return postulation;
```
Comment (cristian-chavez-jalafund) [inline src/dal/Jalasoft.TeamUp.Resumes.DAL/PostulationsRepository.cs]: question: is this a global variable? why is directly returned and in other methods is retrieved with a DB query?

File: src/api/Jalasoft.TeamUp.Resumes.API/Controllers/PatchPostulation.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        public PatchPostulation(IPostulationsService postulationsService)
+        {
+            this.postulationsService = postulationsService;
+        }
+
+        [FunctionName("PatchPostulation")]
+        [OpenApiOperation(operationId: "PatchPostulation", tags: new[] { "Postulations" })]
+        [OpenApiParameter(name: "id", In = ParameterLocation.Path, Required = true, Type = typeof(int), Description = "The postulation identifier.")]
+        [OpenApiRequestBody("application/json", typeof(JsonPatchDocument<Postulation>), Description = "JSON request body containing Postulation")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(Postulation), Description = "Successful response")]
```
Comment (cristian-chavez-jalafund) [inline src/api/Jalasoft.TeamUp.Resumes.API/Controllers/PatchPostulation.cs:33]: is this the only probable response this endpoint will return? or are we missing document others? 😉

**Review by Raul-Gamarra-jalafund — DISMISSED** [review]

File: src/api/Jalasoft.TeamUp.Resumes.API/Controllers/PatchPostulation.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+                return new OkObjectResult(result);
+            }
+            catch (ResumesException e)
+            {
+                return e.Error;
+            }
+            catch (ValidationException exVal)
+            {
+                var errorException = new ResumesException(ResumesErrors.BadRequest, exVal);
```
Comment (cristian-chavez-jalafund) [inline src/api/Jalasoft.TeamUp.Resumes.API/Controllers/PatchPostulation.cs]: should this be documented?

**Review by cristian-chavez-jalafund — APPROVED** [review]

**Review by lina-ramirez-jalafund — APPROVED** [review]

**Review by alex-flores-jalafund — APPROVED** [review]

**Pull request merged**